src/modules/detectChainings.js

```diff
@@ -0,0 +1,73 @@
+import rects from './createRects';
+
+/**
+ * Chain rectangles
+ */
+export default function detectChainings() {
+  const maxDist = 10;
+  const minDist = 0;
+  const distance = (rect1, rect2) => {
+    return Math.sqrt((rect2.x - rect1.x) ** 2 + (rect2.y - rect1.y) ** 2);
+  };
+
+  const currentRect = rects.find((rect) => rect.isDragging);
+  const closestRect = rects.find((rect) => {
+    const distX = parseInt(distance(currentRect, rect) - rect.width, 10);
+    const distY = parseInt(distance(currentRect, rect) - rect.height, 10);
+
+    if ((distX >= minDist && distX <= maxDist) || (distY >= minDist && distY <= maxDist)) {
+      return rect;
+    }
+
+    return null;
+  });
```

Comment:
Lines 14-23 looks like separate function and can be extract from function `detectChainings`
Also I think we can simplify `return` condition.

```
   retrun (distX >= minDist && distX <= maxDist) || (distY >= minDist && distY <= maxDist) ? rect : null;
```